fidesctl/src/fidesctl/cli/__init__.py

```diff
@@ -75,11 +77,17 @@ def cli(ctx: click.Context, config_path: str, local: bool) -> None:
             cli.add_command(command)
     else:
         config.cli.local_mode = True
-    ctx.obj["CONFIG"] = config
 
+    ctx.obj["CONFIG"] = config
     if not ctx.invoked_subcommand:
         click.echo(cli.get_help(ctx))
 
+    # Check the server health and version if an API command is invoked
+    if ctx.invoked_subcommand in [
+        command.name for command in API_COMMANDS if command.name != "webserver"
+    ]:
+        check_server(VERSION, config.cli.server_url)
```

Comment:
I'd prefer if this logic remained in a dedicated command, like `fidesctl status`, for three reasons:
1. Placing it here means that every CLI command that gets executed will first have to make two API calls (one call if my above comment is accepted) before executing the command handler. We've already increased the latency of the CLI significantly with the analytics logic (optimizations TBD), and I don't think it's necessary to do more of that here.
2. I don't believe it's necessary to check the server's health and version before every command. I think of this as more of a debugging feature, and as such it should remain out of the way when not needed. It's fair to argue that we should always make the user aware if their CLI version is mismatched with the API server version. My response would be: if that's the case and still nothing breaks, why slow them down? If that's the case and something _does_ break, then `fidesctl status` is available to them as a debugging tool.
3. Removing this logic means we can revert the change to exclude this file from the cyclomatic complexity check. If we're going to keep feature-packing this function, that check becomes really important.

Reply:
1. I don't think the latency of a single (after consolidation) API call without a db check will actually perceptibly increase the latency of a command. But to prove my point I ran some benchmarks

Simplest possible command
```
root@6fd2c3e84bd3:/fides/fidesctl# time fidesctl
Usage: fidesctl [OPTIONS] COMMAND [ARGS]...

  The parent group for the Fidesctl CLI.

Options:
  --version               Show the version and exit.
  -f, --config-path TEXT  Path to a configuration file. Use 'fidesctl view-   
                          config' to print the config. Not compatible with the
                          'fidesctl webserver' subcommand.

  --local                 Run in 'local_mode'. This mode doesn't make API     
                          calls and can be used without the API
                          server/database.

  -h, --help              Show this message and exit.

Commands:
  annotate   Annotate fidesctl resource types
  apply      Validates local manifest files and then sends them to the...     
  db         Database utility commands
  delete     Delete a resource on the server.
  evaluate   Compare your System's Privacy Declarations with your...
  export     Export fidesctl resource types
  generate   Generate fidesctl resource types
  get        View a resource from the server as a JSON object.
  init       Initializes a Fidesctl instance, creating the default
             directory...

  ls         Get a list of all resources of this type from the server and...  
  parse      Reads the resource files that are stored in MANIFESTS_DIR and... 
  scan       Scan external resource coverage against fidesctl resources       
  view       View various resources types.
  webserver  Starts the fidesctl API server using Uvicorn on port 8080.       

real    0m2.331s
user    0m0.747s
sys     0m0.311s
```

Actually kinda does something
```
root@6fd2c3e84bd3:/fides/fidesctl# time fidesctl --version
fidesctl, version 1.4.2+21.g8e81be0.dirty

real    0m2.285s
user    0m0.723s
sys     0m0.280s
```

Calls the Server and makes a database call
```
root@6fd2c3e84bd3:/fides/fidesctl# time fidesctl ls dataset
[]

real    0m2.613s
user    0m0.787s
sys     0m0.297s
```

The user experience of knowing _immediately_ that your server is broken (proactive debugging) is much better than having something break and assuming our user knows that that command exists and then calling it.

2. See above. Although I totally agree having an optional `fidesctl status` command could be useful for the sake of it

3. There are probably better ways to simplify that function than removing functionality.